### PR TITLE
GetRewardEst simplify

### LINKED ISSUE #109: GetRewardEst iterates all over the locked tokens
`GetRewardsEst`(which is called in `RewardsEst` grpc query) calls `FilteredLocksDistributionEst`, which iterates all over the locked tokens(https://github.com/osmosis-labs/osmosis/blob/main/x/incentives/keeper/pot.go#L164). This causes issues on performance side, as it need to recalculate the reward every time when the query happens.

The suggested solution is limit the lockup period to the epoch, and make the lockup module updates the total amount when there is lock related event happens. The incentive module will read the cached state when queried. 

ref @Thunnini

### PR DESCRIPTION
**Pull request by mconcat**
closes #109

### CODE REVIEW

**Comment by ValarDragon** [comment]
Nice start on this! BTW, I think you're editor doesn't have `gofmt` or `goimports` running on save. (Or if it does, its currently configured to use two spaces for indentation, not four spaces)

File: x/incentives/keeper/lockup.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		x := k.getTotalLockedDenom(ctx, coin.Denom)
+		k.setTotalLockedDenom(ctx, coin.Denom, x.Add(coin.Amount))
+	}
+}
+
+func (k Keeper) DecreaseTotalLocked(ctx sdk.Context, amount sdk.Coins) {
+	for _, coin := range amount {
+		x := k.getTotalLockedDenom(ctx, coin.Denom)
+		k.setTotalLockedDenom(ctx, coin.Denom, x.Sub(coin.Amount))
+		// XXX: invariant no negative
```
Comment (ValarDragon) [inline x/incentives/keeper/lockup.go:40]: Thanks for flagging this place for an invariant! Lets open this up into a prelaunch issue once the PR is merged. (We need to do a lot of work on the simulator / invariants prelaunch)

**Comment by ValarDragon** [comment]
btw is this ready for review?

File: x/incentives/keeper/lockup.go (hunk 1 above)
Comment (mconcat) [reply x/incentives/keeper/lockup.go:40]: Sounds good!

Comment (sunnya97) [reply x/incentives/keeper/lockup.go:40]: Opened #140

**Comment by mconcat** [comment]
Closing on behalf of #156

**Pull request closed without merging**